File: StarLightCore/src/Composestar/Core/INLINE/model/ContextExpression.java
/*
 * Created on 27-sep-2006
 *
 */
package Composestar.Core.INLINE.model;


public class ContextExpression{
    private int type;
    
    public final static int HAS_MORE_ACTIONS = 22;
    public final static int RETRIEVE_ACTION = 23;
    
    public ContextExpression( int type ){
        this.type = type;
    }

    /**
     * @return the type
     */
    public int getType(){
        return type;
    }
    
    
    
}

File: Source/StarLightEntities/WeaveSpec/Instructions/ContextExpression.cs
#region Using directives
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;
using System.Xml.Serialization;
using Composestar.StarLight.Entities.WeaveSpec.Instructions.Visitor;
#endregion

/// <summary>
/// Composestar. star light. weave spec. instructions
/// </summary>
namespace Composestar.StarLight.Entities.WeaveSpec.Instructions
{

    /// <summary>
    /// This expression determines the type of code to be injected for a context expression.
    /// </summary>
    [Serializable]
    [XmlRoot("ContextExpression", Namespace = "Entities.TYM.DotNET.Composestar")]
    public enum ContextExpression
    {
        /// <summary>
        /// Checks if there are more actions in the action store.
        /// </summary>
        HasMoreActions = 22,
        /// <summary>
        /// Retrieves the next action in the action store.
        /// </summary>
        RetrieveAction = 23        
    } // enum ContextExpression

} // namespace Composestar.StarLight.Entities.WeaveSpec.Instructions
